scripts/results_verification.py

```diff
@@ -0,0 +1,74 @@
+import sys
+import os
+from pathlib import Path
+import xml.etree.ElementTree as ET
+
+
+def check_file_size(file_path, max_size=5):
+    """
+    Check if the file size of a given file is smaller than max_size.
+
+    :param file_path: path of junit xml report from sanitycheck to be cheched
+    :param max_size: max size allowed for the given file
+    """
+    file_size = file_path.stat().st_size / 1000000
+    return bool(file_size <= max_size)
+
+
+def check_item_count(file_path=None, item=None, max_count=None):
+    """
+    Check if the count of a given item is smaller than max_count. Item can be e.g. 'errors' or 'failures'
+
+    :param file_path: path of junit xml report from sanitycheck to be cheched
+    :param item: item to be verified
+    :param max_count: max amount of the given item
+    """
+    testsuite = ET.parse(file_path).getroot()
+    for summary in testsuite:
+        item_count = int(summary.attrib[item])
+        return bool(item_count <= max_count)
+
+
+def check_version(file_path=None, version=None):
+    """
+    Check if zephyr's version given in the report match the expected one
+
+    :param file_path: junit xml report from sanitycheck
+    :param version: expected version of zephyr
+    """
+    testsuite = ET.parse(file_path).getroot()[0]
+    properties = testsuite[0]
+    for property in properties:
+        if property.attrib['name'] == "version":
+            return bool(property.attrib['value'] == version)
+
+    print("Version not found.")
+    return False
+
+
+def main(path, version):
+    # TODO: add argparse
+    max_size = 5
+    max_failures = 50
+    max_errors = 50
+
+    file_path = Path(path)
+
+    if not os.path.exists(file_path):
```

Comment:
I think your idea won't work with the publishing workflow. In a single folder (corresponding to a given zephyr version) there will be reports from all platforms, as one adds their results to this folder. The script would then run on the same files multiple times with each commit targeting the same version. IMO it is cleaner if some higher layer decides which file to verify and this could be done in github actions by scanning the xml files which are updated in the PR.

Reply:
GHAction check file one by one, but if I want to try it on my local first, I need run for multiply times.